### PR TITLE
Add circle-ci configs to repo

### PR DESCRIPTION
**Pull request by briancain**
This commit adds some circle-ci automation to lint and run the unit
tests

### CODE REVIEW

File: .circleci/bash_env.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+export GIT_COMMIT=$(git rev-parse --short HEAD)
+export GIT_DIRTY=$(test -n "`git status --porcelain`" && echo "+CHANGES" || true)
+export GIT_DESCRIBE=$(git describe --tags --always --match "v*")
+export GIT_IMPORT=github.com/hashicorp/waypoint/internal/version
```
Comment (briancain) [inline .circleci/bash_env.sh:6]: Not sure if we should copy this over at all from waypoint or just remove this script completely :smile: mostly just hoping circle can run these unit tests per pull request so maybe this isn't required here.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            gotestsum --format=short-verbose \
+              --junitfile $TEST_RESULTS_DIR/gotestsum-report.xml -- \
+              -tags="$GOTAGS" -p 2 \
+              -cover -coverprofile=coverage.txt \
+              $PACKAGE_NAMES
+      - store_test_results:
+          path: *TEST_RESULTS_DIR
+      - store_artifacts:
+          path: *TEST_RESULTS_DIR
+      - notify_main_failure
```
Comment (krantzinator) [inline .circleci/config.yml:143]: This is a step defined by us, not something built in to circle. https://github.com/hashicorp/waypoint/blob/a2bc8b461adace6c720645a02d7c7ca26730912a/.circleci/config.yml#L509-L550
We'll need to make sure this project has a `SLACK_WEBHOOK` defined as well.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+              nc -z localhost 5432 && echo Success && exit 0
+              echo -n .
+              sleep 1
+            done
+            echo Failed waiting for Postgres && exit 1
+      - run:
+          name: go test
+          command: |
+            mkdir -p $TEST_RESULTS_DIR
+            PACKAGE_NAMES=$(go list ./... | circleci tests split --split-by=timings --timings-type=classname)
```
Comment (krantzinator) [inline .circleci/config.yml:131]: I'm getting an issue with not being able to run the `windows` specific stuff locally. Have you found a way around this?
```
package github.com/hashicorp/waypoint-plugin-sdk/internal/pkg/conpty
        imports golang.org/x/sys/windows: build constraints exclude all Go files in /home/rae/go/pkg/mod/golang.org/x/sys@v0.0.0-20200916030750-2334cc1a136f/windows
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        command: ["pebble"]
+        environment:
+          PEBBLE_VA_NOSLEEP: 1
+          PEBBLE_VA_ALWAYS_VALID: 1
+      - image: docker.mirror.hashicorp.services/circleci/postgres:11-alpine
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: waypoint_test
+      - image: docker.mirror.hashicorp.services/vault
+        command: ["server", "-dev", "-dev-root-token-id=hznroot"]
```
Comment (krantzinator) [inline .circleci/config.yml:90]: I don't htink we need to worry about postgres or horizon on the sdk side?

**Review by krantzinator — CHANGES_REQUESTED** [review]
Thanks for doing this! Couple of questions and changes.

File: .circleci/bash_env.sh (hunk 1 above)
Comment (krantzinator) [reply .circleci/bash_env.sh:6]: I agree we don't need all of this

File: .circleci/config.yml (hunk 4 above)
Comment (briancain) [reply .circleci/config.yml:90]: Yep agreed....I missed some stuff from the copy over from Waypoint! :smile:

File: .circleci/config.yml (hunk 3 above)
Comment (briancain) [reply .circleci/config.yml:104]: I have not, although if I remember right I don't think that package works on windows? :thinking: @evanphx @mitchellh

File: .circleci/config.yml (hunk 2 above)
Comment (briancain) [reply .circleci/config.yml:116]: Good catch! Is there a good way to do that? Is it just something we add in circle once the project workflow has been created? :thinking:

File: .circleci/config.yml (hunk 3 above)
Comment (briancain) [reply .circleci/config.yml:103]: Fixed via https://github.com/hashicorp/waypoint-plugin-sdk/pull/18

File: .circleci/config.yml (hunk 2 above)
Comment (briancain) [reply .circleci/config.yml:115]: Ok, we got this sorted out now I think. We created a new webhook and I just added it to the project on Circle :)

**Comment by krantzinator** [comment]
Hm do we have to commit this to `main` before it will run on PRs?

**Comment by briancain** [comment]
Seems like it? I was hoping it would run in here but it doesn't seem like it does, maybe it's a security thing?

**Comment by briancain** [comment]
Approved via :shipit: emoji in slack 😎

**Pull request merged**